pyiceberg/table/inspect.py

```diff
@@ -157,74 +158,96 @@ def _readable_metrics_struct(bound_type: PrimitiveType) -> pa.StructType:
                 pa.field("readable_metrics", pa.struct(readable_metrics_struct), nullable=True),
             ]
         )
+        return entries_schema
+
+    def _get_entries(self, schema: "pa.Schema", manifest: ManifestFile, discard_deleted: bool = True) -> "pa.Table":
```

Comment:
schema here should be iceberg schema instead of pa.schema

Reply:
there are still a couple places in this function using `self.tbl.` which is the table's current metadata
L189 & L195.
We should use the schema and partition spec at the time of the snapshot instead